controllers/topics/index.js

```diff
@@ -204,11 +212,31 @@ const followTopicV2 = async (req, res) => {
           detailTokenUser.user_id,
           name,
           detailTokenUser.is_anonymous,
-          with_system_message,
-          prevUserToken
+          topicInvitationIds,
+          prevUserToken,
+          with_system_message
         )
       );
     } else {
+      //get invitations topic detail
+      const topicInvitations = await TopicInvitations.findAll({
+        where: {
+          user_id_invited: userId,
+          topic_id
+        }
+      });
+
+      if (topicInvitations.length > 0) {
+        topicInvitationIds = topicInvitations.map(
+          (topicInvitation) => topicInvitation.topic_invitations_id
+        );
+        await TopicInvitations.destroy({
+          where: {
+            topic_invitations_id: topicInvitationIds
+          }
+        });
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code handles the case where a user is not following a topic. It fetches all topic invitations for the user and the specific topic, stores their IDs in `topicInvitationIds`, and then deletes these invitations. This is a good way to clean up any pending invitations when a user decides to follow a topic.
```diff
+      //get invitations topic detail
+      const topicInvitations = await TopicInvitations.findAll({
+        where: {
+          user_id_invited: userId,
+          topic_id
+        }
+      });
+
+      if (topicInvitations.length > 0) {
+        topicInvitationIds = topicInvitations.map(
+          (topicInvitation) => topicInvitation.topic_invitations_id
+        );
+        await TopicInvitations.destroy({
+          where: {
+            topic_invitations_id: topicInvitationIds
+          }
+        });
+      }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->